openquake/commonlib/commands/run.py

```diff
@@ -16,15 +16,45 @@
 #  You should have received a copy of the GNU Affero General Public License
 #  along with OpenQuake.  If not, see <http://www.gnu.org/licenses/>.
 from __future__ import print_function
+import collections
 import logging
 import cProfile
+import pstats
+import io
 
 from openquake.baselib import performance, general
 from openquake.commonlib import sap, readinput, valid, datastore
-from openquake.calculators import base
+from openquake.calculators import base, views
 
 calc_path = None  # set only when the flag --profile is given
 
+PStatData = collections.namedtuple(
+    'PStatData', 'ncalls tottime percall cumtime percall2 path')
+
+
+def get_pstats(pstatfile, n):
+    """
+    Return profiling information as an RST table.
+
+    :param pstatfile: path to a .pstat file
+    :param n: the maximum number of stats to retrieve
+    """
+    stream = io.BytesIO()
+    ps = pstats.Stats(pstatfile, stream=stream)
+    ps.sort_stats('cumtime')
+    ps.print_stats(n)
+    lines = stream.getvalue().splitlines()
+    for i, line in enumerate(lines):
+        if line.startswith('   ncalls'):
```

Comment:
It would be more readable if you could add a small comment with an example of what you expect to read (useful to understand here and line 51)